src/query.c

```diff
@@ -646,9 +646,27 @@ static IndexIterator *Query_EvalTagPrefixNode(QueryEvalCtx *q, TagIndex *idx, Qu
   return NewUnionIterator(its, itsSz, q->docTable, 1, weight, QN_PREFIX, qn->pfx.str);
 }
 
+static void tag_strtolower(char *str, size_t *len) {
+  char *p = str;
+  while (*p) {
+    if (*p == '\\') {
+      ++p;
+      --*len;
```

Comment:
Don't we already have code that does tolower and remove escape character, so we can avoid another version here?
Better to decrease a local variable than a pointer? (to avoid dereferencing)

Reply:
We saw that `strdupcase` is not exactly what is needed here (not updating `len`, handling not only baskslach, etc.) so we can remain with the initial code.